src/components/molecules/Carousel/Carousel.tsx

```diff
@@ -0,0 +1,123 @@
+import { Box } from '@mui/material'
+import React, { createRef, RefObject, useEffect, useState } from 'react'
+import './carousel.scss'
+
+export type CarouselProps = {
+  elementWidth: number
+  elementHeight: number
+  selfRotate: boolean
+  elements: React.ReactNode[]
+}
+
+const defaultCarouselProps = {
+  elementWidth: 210,
+  elementHeight: 140,
+  selfRotate: true,
+  elements: [],
+}
+
+export const Carousel = (props: CarouselProps = defaultCarouselProps) => {
+  const [selectedCell, setSelectedCell] = useState(0)
+  const [intervalId, setIntervalId] = useState<NodeJS.Timer | null>(null)
+
+  const carouselRef: RefObject<HTMLDivElement> = createRef<HTMLDivElement>()
+  // Refs to move the elemenets
+  const elementRefs: RefObject<HTMLDivElement>[] = props.elements.map(() => createRef<HTMLDivElement>())
+
+  function getHelperData() {
+    const cellCount = props.elements.length
+    const theta = 360 / cellCount
+    const cellSize = props.elementWidth + 16
+
+    const radius = Math.round(cellSize / 2 / Math.tan(Math.PI / cellCount))
+
+    return { cellCount, theta, radius, cellSize }
+  }
+
+  function rotateCarousel() {
+    const { theta, radius } = getHelperData()
+    const angle = theta * selectedCell * -1
```

Comment:
I know it's complicate but would be possible to add a comment to explain the logic here? in two weeks we will forgot about this

Reply:
I can give a try, its all mathematical stuff